### PR TITLE
Adds simple Exists implementation to machine actuator

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuck@heptio.com>

I'm not sure if this makes sense to merge now, but it's a little bit of code/test implements a very basic existence check

### CODE REVIEW

File: cloud/aws/actuators/machine/actuator_test.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ func (e *ec2) InstanceIfExists(id *string) (*ec2svc.Instance, error) {
 	}
 	if *id == "abc" {
 		return &ec2svc.Instance{
-			ID: "abc",
+			ID:    "abc",
+			State: "running",
```
[the remark this answers is not in the document]
Comment (chuckha) [reply cloud/aws/actuators/machine/actuator_test.go]: yeah. actually i should probably use a constant assuming the ec2 sdk defines instance states as constants.

File: cloud/aws/actuators/machine/actuator.go [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 package machine
 
+// should not need to import the ec2 sdk here
```
Comment (chuckha) [inline cloud/aws/actuators/machine/actuator.go]: Is this excessive?

File: cloud/aws/actuators/machine/actuator_test.go [hunk 3]
```diff
@@ -66,3 +67,43 @@ func TestCreate(t *testing.T) {
...
+func TestExists(t *testing.T) {
+	testCases := []struct {
+		name    string
+		exists  bool
+		machine *clusterv1.Machine
+	}{
+		{
+			name:    "machine does not exist",
+			exists:  false,
+			machine: &clusterv1.Machine{},
```
Comment (detiber) [inline cloud/aws/actuators/machine/actuator_test.go]: Should there be a positive test case here as well?

Comment (chuckha) [reply cloud/aws/actuators/machine/actuator_test.go]: yeah, i think the pattern outlined in my other pr (https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/74) makes this a lot nicer

**Comment by chuckha** [comment]
Test deleted because the actuator test covered the test cases of the instance_test.go file. If we need to specifically test one piece of instance we can do that, but we should prefer the more integration style testing where we use the real actuator and the real service with a mocked out ec2.

File: cloud/aws/actuators/machine/actuator_test.go [hunk 4]
```diff
@@ -16,28 +16,28 @@ package machine_test
...
-	if id == nil {
-		return nil, nil
-	}
-	if *id == "abc" {
-		return &ec2svc.Instance{
-			ID: "abc",
-		}, nil
-	}
+
+func (i *instancesSvc) DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
```
Comment (chuckha) [inline cloud/aws/actuators/machine/actuator_test.go]: will convert to gomock later

**Pull request merged**